File: 1250-check-if-it-is-a-good-array/1250-check-if-it-is-a-good-array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isGoodArray = function(nums) {
    function gcd(a, b) {
        while (b !== 0) {
            let temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    let n = nums.length;
    let gcdivi = nums[0];

    if(gcdivi == 1) return true;

    for(let i = 1; i < n; i++) {
        gcdivi = gcd(nums[i], gcdivi);

        if(gcdivi == 1) return true;
    }
    return false;
};

File: 1250-check-if-it-is-a-good-array/1250-check-if-it-is-a-good-array.cpp
#include <numeric>
class Solution {
public:
    bool isGoodArray(vector<int>& nums) {
        int gcdivi = nums[0];

        for (int num : nums) {
            gcdivi = gcd(gcdivi, num);

            if (gcdivi == 1)
                return true;
        }
        return false;
    }
};
